### PR TITLE
Restore pause on macOS & pause indicator in the window's title bar

### PR DESCRIPTION
**Pull request by johnnovak**
# Description

Due to popular demand from @Grounded0 😄 , this restores the pause function on macOS (fixes https://github.com/dosbox-staging/dosbox-staging/issues/1865). The hotkey is hardcoded: Cmd+P pauses, then Cmd+P again unpauses.

It also restores the pause indicator in the window's title bar on all platforms, so at least in windowed mode you'll know DOSBox is not responding because it's paused:

image

---

To recap:

- Pause/unpause is hardcoded to Alt+Pause on Windows & Linux.
- Using Esc to unpause caused issues, so that has been removed.
- If you remap the "Pause Emu." hotkey in the mapper, you'll get all sorts of erratic behaviour—don't do it. Remapping the pause key is currently not supported.
- User input is effectively stopped in Pause mode as it's a hack. So you cannot use any other hotkeys like creating a screenshot while the game is paused, or enter/exit fullscreen, or anything else. All you can do is exit pause mode. More info: https://github.com/dosbox-staging/dosbox-staging/pull/2323#issuecomment-1462349376

The pause feature will be rebuilt and properly implemented when we get to reimplementing the mapper from the ground up, but most likely not until then as the current implementation is a royal mess...

## Related issues

https://github.com/dosbox-staging/dosbox-staging/issues/1865

https://github.com/dosbox-staging/dosbox-staging/pull/1970

# Manual testing

- Pause/unpause works in windowed mode and you can see the "PAUSED" indicator in the window's title bar
- Pause/unpause works in fullscreen mode
- Cmd+Q or clicking on the red close window titlebar widget exists DOSBox while in pause mode
- Minimising the paused window then restoring it works
- If `pause_when_inactive` is enabled, manual pause overrides the auto-unpause (so if you've paused with Cmd+P, deactivating and activating the window again won't unpause—I think this is fine and logical)
 
# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

**Comment by Grounded0** [comment]
image

**Comment by Grounded0** [comment]
Pinging our Quality Assurance Team.

@Burrito78

**Comment by Grounded0** [comment]
Since it's a hardcoded key let me justify why `⌘ Command-P`:

Doesn't interfere with DOS or any of macOS Core Services functions. Interferes with macOS HIG Print... function but since we have professional grade printing through PostScript it doesn't matter for now.

If line printing and Epson ESC/P printing is added maybe that key is assigned somewhere else OR we've managed to make Pause key reassignable again. This is an interim fix for now to restore functionality.

**Comment by johnnovak** [comment]
I think Cmd+P is a great choice @Grounded0.

**Comment by Grounded0** [comment]
After reassigning Pause function in keymapper to Command-P seems to function as stated.

**Review by Grounded0 — APPROVED** [review]

**Comment by Grounded0** [comment]
Actually this closes another ticket too:

-#2632

**Comment by johnnovak** [comment]
> After reassigning Pause function in keymapper to Command-P seems to function as stated.

Yeah please mention it in the release notes that assigning Pause to anything else than Cmd+P will screw the feature up.

**Comment by Burrito78** [comment]
Works for me as intended! There is a slight pause before it kicks in though. Think that's by design?

**Comment by johnnovak** [comment]
> Works for me as intended! There is a slight pause before it kicks in though. Think that's by design?

No, it just work like that... dunno why! I'm glad it works at all 😅

**Comment by Grounded0** [comment]
Waiting for @kcgen to clear this and push big green merge button.

**Comment by johnnovak** [comment]
Between the three of testing this on macOS, I think this is safe to merge 😄

**Pull request merged**
